client/my-sites/plugins/main.jsx

```diff
@@ -79,12 +83,14 @@ export class PluginsMain extends Component {
 			search,
 		} = this.props;
 
-		currentPlugins.map( ( plugin ) => {
-			const pluginData = this.props.wporgPlugins?.[ plugin.slug ];
-			if ( ! pluginData ) {
-				this.props.wporgFetchPluginData( plugin.slug );
-			}
-		} );
+		if ( currentPlugins.length <= WPORG_PLUGIN_COUNT_CUTOFF ) {
+			currentPlugins.forEach( ( plugin ) => {
+				const pluginData = this.props.wporgPlugins?.[ plugin.slug ];
+				if ( ! pluginData ) {
+					this.props.wporgFetchPluginData( plugin.slug );
+				}
+			} );
+		}
```

Comment:
This was intended to be a temporary fix, you can read some of the discussion on it here: p1671076313369439-slack-C039K7CD119. The 20 number is definitely arbitrary.

I am partial to the method of caching these results on the server so that they are all retrieved in a single request but I also recognize that this is a lot of dev effort.

I moved these changes to https://github.com/Automattic/wp-calypso/pull/72621 so that they aren't in this PR.

Reply:
Thanks for moving them out - feel free to ping me when this PR is ready for review and we can discuss it separately.